src/ContractionTreeToGraph.jl

```diff
@@ -0,0 +1,120 @@
+
+"""The main object here is `g' a NamedGraph which represents a graphical version of a contraction sequence.
+It's vertices describe a partition between the leaves of the sequence (will be labelled with an n = 1 or n = 3 element tuple, where each element of the tuple describes the leaves in one of those partition)
+n = 1 implies the vertex is actually a leaf.
+Edges connect vertices which are child/ parent and also define a bi-partition"""
+
+
+"""Function to take a sequence (returned by ITensorNetworks.contraction_sequence) and construct a graph g which represents it (see above)"""
+function create_contraction_tree(contract_sequence)
+    #Collect all the leaves
+    leaves = collect(Leaves(contract_sequence))
+    #Build empty graph
+    g = NamedGraph()
+    #Construct the left branch
+    spawn_child_branch(contract_sequence[1], g, leaves)
```

Comment:
I would call this:
```julia
fill_contraction_sequence_graph_vertices!(g, contract_sequence)
```
and then have it organized like:
```julia
function fill_contraction_sequence_graph_vertices!(g, contract_sequence)
  leaves = collect(Leaves(contract_sequence))
  fill_contraction_sequence_graph_vertices!(g, contract_sequence, leaves)
  return g
end

function fill_contraction_sequence_graph_vertices!(g, contract_sequence, leaves)
  fill_contraction_sequence_graph_vertices!(g, contract_sequence[1], leaves)
  fill_contraction_sequence_graph_vertices!(g, contract_sequence[2], leaves)
  return g
end
```
or something like that. But the point would be to hide more details away in layers of functions and out of the top level `create_contraction_tree`/`contraction_sequence_to_graph` function.